packages/web-components/src/tablist/tablist.ts

```diff
@@ -0,0 +1,438 @@
+    } else {
+      return 0;
+    }
+  }
+
+  /**
+   * Function that is invoked whenever the selected tab or the tab collection changes.
+   *
+   * @public
+   */
+  protected setTabs(): void {
+    this.activeTabIndex = this.getActiveIndex();
+
+    this.tabs.forEach((tab: HTMLElement, index: number) => {
+      if (tab.slot === 'tab') {
+        const isActiveTab = this.activeTabIndex === index && this.isFocusableElement(tab);
+
+        const tabId: string = this.tabIds[index];
+        tab.setAttribute('id', tabId);
+        tab.setAttribute('aria-selected', isActiveTab ? 'true' : 'false');
+        tab.addEventListener('click', this.handleTabClick);
+        tab.addEventListener('keydown', this.handleTabKeyDown);
+        tab.setAttribute('tabindex', isActiveTab ? '0' : '-1');
+        if (isActiveTab) {
+          this.activetab = tab;
+          this.activeid = tabId;
+        }
+      }
+    });
+  }
+
+  private getTabIds(): Array<string> {
+    return this.tabs.map((tab: HTMLElement) => {
+      return tab.getAttribute('id') ?? `tab-${uniqueId()}`;
+    });
+  }
+
+  private setComponent(): void {
+    if (this.activeTabIndex !== this.prevActiveTabIndex) {
+      this.activeid = this.tabIds[this.activeTabIndex] as string;
+      this.focusTab();
+      this.change();
+    }
+  }
+
+  private handleTabClick = (event: MouseEvent): void => {
+    const selectedTab = event.currentTarget as HTMLElement;
+    if (selectedTab.nodeType === 1 && this.isFocusableElement(selectedTab)) {
+      this.prevActiveTabIndex = this.activeTabIndex;
+      this.activeTabIndex = this.tabs.indexOf(selectedTab);
+      this.setComponent();
+    }
+  };
+
+  private isHorizontal(): boolean {
+    return this.orientation === TablistOrientation.horizontal;
+  }
+
+  private handleTabKeyDown = (event: KeyboardEvent): void => {
+    if (this.isHorizontal()) {
+      switch (event.key) {
+        case keyArrowLeft:
+          event.preventDefault();
+          this.adjustBackward(event);
+          break;
+        case keyArrowRight:
+          event.preventDefault();
+          this.adjustForward(event);
+          break;
+      }
+    } else {
+      switch (event.key) {
+        case keyArrowUp:
+          event.preventDefault();
+          this.adjustBackward(event);
+          break;
+        case keyArrowDown:
+          event.preventDefault();
+          this.adjustForward(event);
+          break;
+      }
+    }
+    switch (event.key) {
+      case keyHome:
+        event.preventDefault();
+        this.adjust(-this.activeTabIndex);
+        break;
+      case keyEnd:
+        event.preventDefault();
+        this.adjust(this.tabs.length - this.activeTabIndex - 1);
+        break;
+    }
+  };
+
+  /**
+   * The adjust method for FASTTabs
+   * @public
+   * @remarks
+   * This method allows the active index to be adjusted by numerical increments
+   */
+  public adjust(adjustment: number): void {
```

Comment:
Do we need `adjust`, `adjustForward` and `adjustBackward`. can the latter be achieved by calling `adjust(1)` and `adjust(-1)`?